app/javascript/gobierto_data/webapp/components/sets/VisualizationsItem.vue

```diff
@@ -105,83 +113,85 @@ export default {
       type: Boolean,
       default: false
     },
-    isPrivateVizLoading: {
+    showPrivatePublicIconViz: {
       type: Boolean,
       default: false
     },
+    showPrivateViz: {
+      type: Boolean,
+      default: false
+    },
+    showPrivate: {
+      type: Boolean,
+      default: false
+    },
+    showLabelEdit: {
+      type: Boolean,
+      default: false
+    },
+    isVizItemModified: {
+      type: Boolean,
+      default: false
+    },
+    resetPrivate: {
+      type: Boolean,
+      default: false
+    }
   },
   data() {
     return {
       labelVisName: I18n.t('gobierto_data.projects.visName') || "",
-      labelEdit: I18n.t('gobierto_data.projects.edit') || "",
-      labelSaveViz: I18n.t('gobierto_data.projects.saveViz') || "",
       labelVisualize: I18n.t('gobierto_data.projects.visualize') || "",
       labelDashboard: I18n.t('gobierto_data.projects.dashboards') || "",
       labelSavedVisualization: I18n.t("gobierto_data.projects.savedVisualization") || "",
       labelModifiedVizualition: I18n.t("gobierto_data.projects.modifiedVisualization") || "",
+      labelQuery: I18n.t("gobierto_data.projects.query") || "",
       items: null,
       config: {},
       vizID: null,
+      queryID: '',
+      queryName: '',
       user: null,
       queryViz: '',
       isVizElementSavingVisible: false,
       name: '',
-      isQuerySavingPromptVisible: false,
-      tabs
-    }
-  },
-  computed: {
-    checkVisualizationsItems() {
-      return (this.privateVisualizations.length && this.items) || (this.publicVisualizations.length && this.items)
+      isQuerySavingPromptVisible: false
     }
   },
   watch: {
-    publicVisualizations(newValue, oldValue) {
-      if (newValue !== oldValue) {
-        this.getDataVisualization(newValue);
-      }
-    },
-    privateVisualization(newValue, oldValue) {
-      if (newValue !== oldValue) {
-        this.getDataVisualization(newValue);
-      }
-    },
     vizInputFocus(newValue) {
       if (newValue) {
         this.$nextTick(() => this.$refs.savingDialogVizElement.inputFocus())
       }
-    },
-    async $route(to, from) {
-      if (to.path !== from.path) {
-        await this.getDataVisualization(this.privateVisualizations);
-        await this.getDataVisualization(this.publicVisualizations);
-      }
     }
   },
-  async created() {
+  created() {
+    this.$root.$emit("isVizModified", false);
```

Comment:
I don't see the point of calling to this events both in created and destroy hooks. Why a component should do these sort of things when there's no user interaction? Is really so necessary?

Reply:
It's not necessary